feat: Add --map-root argument to configure repository map root directory

coredevorg opened the pull request:
The patch makes the root directory for `.aider.tags.cache.v3` configurable through an additional argument `--map-root` or `AIDER_MAP_ROOT`environment variable.

This becomes important if your project lives on a CIFS mount and you encounter:

```
Tags cache error: database is locked
Unable to use tags cache at /mnt/.../.aider.tags.cache.v3, falling back to memory cache
Cache recreation error: [Errno 11] Resource temporarily unavailable: 'cache.db'
```

With the `--map-root` parameter, the cache directory can be moved to local device to avoid the problem.

coredevorg commented:
Yes, you are right @Finndersen, it's the directory where the repomap cache _.aider.tags.cache.v3_ is located and has nothing to do with the repository itself. The option is related to an existing option group [Repomap settings](https://aider.chat/docs/config/options.html#repomap-settings). So I'm fine with _--map-root_ in this context, but if you have a better name for the option, let me know...

Finndersen commented:
Maybe something like `--map-cache-dir` would be more appropriate to avoid ambiguity

coredevorg commented:
Changed the argument name according to a suggestions from @Finndersen to less ambiguous --map-cache-dir

Finndersen commented:
@coredevorg i still `--map-root` is misleading because this change is about the map cache file location, not the root directory of the repo map itself. 
How about `--map-cache-root`? Although "root" is often used to refer to a base directory which is expected to contain subdirectories. If it's just a directory for a single file to be located in, then "dir" is probably more appropriate

coredevorg commented:
Sorry for the confusion, @Finndersen. Unfortunately it's not just the name of the argument in args.py and main.py but the variable is used via **kwargs in the other modules. My quick-and-dirty hack to change the name of the argument has broken the patch so I have to revert because I need the patched aider in my environment. If you really want to change the name, it needs a more in depth refactoring and testing. I won't be able to take a closer look at it for at least 2 weeks. And, btw, it's really the root directory for the cache subdirectory _.aider.tags.cache.v3_ containing the _cache.db_.

Finndersen commented:
Ok, well I still think the user interface/API design should probably be priority. Would `--map-cache-root` make sense? Just because theoretically there could be an argument/param for configuring the root of the repo map itself, so I think it would be backing ourselves into a corner if we used `--map-root` for this

nktl commented:
Certainly a very much needed improvement, right now I am getting harried by stream of "Unable to use tags cache at" errors for various reasons (likely related to 'bad' default location) - and it looks like the memory cache does not persist between queries at all, Aider is recomputing it for every request.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Aider-AI/aider?pullRequest=2911) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Aider-AI/aider?pullRequest=2911) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Aider-AI/aider?pullRequest=2911) it.</sub>

coredevorg commented:
Fixed merge conflicts to be compatible with v0.81.1 and updated sources with a clean commit.

coredevorg commented:
Hi @paul-gauthier , any chance to get these minor (but important for some environments) enhancements merged? Let me know, if I can help in any way...

coredevorg commented:
fix: moved the --map-root argument into the correct root "Repomap settings"

will-rest commented:
Can't wait for this to be merged!

Quick question @coredevorg , where would you recommend to configure `.aider.tags.cache.v3` path. Would it be your home directory?

coredevorg commented:
I'm using a a directory structure like
```
Develop/
├── aider-map-roots
│   └── project
│       └── .aider.tags.cache.v4
│           └── cache.db
└── cifs-mount
    └── project
        ├── .aider.chat.history.md
        ├── .aider.conf.yml
        ├── .aider.input.history
        ├── .git
        └── .vscode```
and an configuration `.aider.conf.yml` with
```
map-root: ~/Develop/aider-map-roots/project
```

coredevorg commented:
Hi @will-rest , following the suggestion of @Finndersen I changed the argument name to `--map-cache-dir` and updated the PR with a clean commit. Maybe this will accelerate approval and merge.
